File: 백준/Silver/1193. 분수찾기/분수찾기.cc
#include <iostream>
using namespace std;

int main()
{
	int x = 0;
	int y = 0;
	int input;
	cin >> input;
	int increase = -1;
	bool check = false;
	
	for (int i = 0; i < input; i++)
	{
		if (x == 0 && y == 0)
		{
			x++;
			y++;
			continue;
		}
		if (x == 1 && !check)
		{
			y++;
			increase *= -1;
			check = true;
			continue;
		}
		if (y == 1 && !check)
		{
			x++;
			increase *= -1;
			check = true;
			continue;
		}

		check = false;
		x += increase;
		y -= increase;
	}
	cout << x << '/' << y;
}

File: 백준/Silver/1193. 분수찾기/분수찾기.py
x = int(input())
max = 0
i, j = 0, 0
while max < x:
    max = (1 + i)*i//2
    i += 1
if(i % 2 == 0):
    print(f'{1+(max-x)}/{i-1-(max-x)}')
else:
    print(f'{i-1-(max-x)}/{1+(max-x)}')